pkg/git/git.go

```diff
@@ -0,0 +1,70 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/log"
+)
+
+
+var (
+	requiredGitHubScopes = []string{ "repo", "admin:repo_hook" }
+)
+
+
+func VerifyToken(ctx context.Context, provider string, token string) (bool, error) {
+	providerToVerifier := map[string]func(context.Context, string)(bool, error){}
+
+	providerToVerifier["github"] = verifyGitHubTokenScope
+	providerToVerifier["gitlab"] = verifyGitLabTokenScope
+
+	verifier := providerToVerifier[provider]
```

Comment:
what if `verifier` is `nil`?

Reply:
it can't be, the installation will fail if the provider is not github or gitlab

Reply:
and you think if tomorrow someone adds bitbucket they will remember to add it here?

Reply:
I think you should only have github here, and any other provider for now should go to a fallback verifier that looks something like the gitlab one.